public/app/plugins/datasource/tempo/traceql/TraceQLEditor.tsx

```diff
@@ -36,73 +36,76 @@ export function TraceQLEditor(props: Props) {
+          },
+          scrollBeyondLastLine: false,
+          wordWrap: 'on',
+        }}
+        onBeforeEditorMount={ensureTraceQL}
+        onEditorDidMount={(editor, monaco) => {
+          if (!props.readOnly) {
+            setupAutocompleteFn(editor, monaco, setupRegisterInteractionCommand(editor));
+            setupActions(editor, monaco, () => onRunQueryRef.current());
+            setupPlaceholder(editor, monaco, styles);
+          }
+          setupAutoSize(editor);
 
-        // Register callback for query changes
-        editor.onDidChangeModelContent((changeEvent) => {
+          // Parse query that might already exist (e.g., after a page refresh)
           const model = editor.getModel();
-
-          if (!model) {
-            return;
+          if (model) {
+            const errorNodes = getErrorNodes(model.getValue());
+            setMarkers(monaco, model, errorNodes);
           }
 
-          // Remove previous callback if existing, to prevent squiggles from been shown while the user is still typing
-          window.clearTimeout(errorTimeoutId.current);
-
-          const errorNodes = getErrorNodes(model.getValue());
-          const cursorPosition = changeEvent.changes[0].rangeOffset;
-
-          // Immediately updates the squiggles, in case the user fixed an error,
-          // excluding the error around the cursor position
-          setMarkers(
-            monaco,
-            model,
-            errorNodes.filter((errorNode) => !(errorNode.from <= cursorPosition && cursorPosition <= errorNode.to))
-          );
-
-          // Later on, show all errors
-          errorTimeoutId.current = window.setTimeout(() => {
-            setMarkers(monaco, model, errorNodes);
-          }, 500);
-        });
-      }}
-    />
+          // Register callback for query changes
+          editor.onDidChangeModelContent((changeEvent) => {
+            const model = editor.getModel();
+
+            if (!model) {
+              return;
+            }
+
+            // Remove previous callback if existing, to prevent squiggles from been shown while the user is still typing
+            window.clearTimeout(errorTimeoutId.current);
+
+            const errorNodes = getErrorNodes(model.getValue());
+            const cursorPosition = changeEvent.changes[0].rangeOffset;
+
+            // Immediately updates the squiggles, in case the user fixed an error,
+            // excluding the error around the cursor position
+            setMarkers(
+              monaco,
+              model,
+              errorNodes.filter((errorNode) => !(errorNode.from <= cursorPosition && cursorPosition <= errorNode.to))
+            );
+
+            // Later on, show all errors
+            errorTimeoutId.current = window.setTimeout(() => {
+              setMarkers(monaco, model, errorNodes);
+            }, 500);
+          });
+        }}
+      />
+      <TemporaryAlert severity={AlertSeverity.Error} text={alertText} />
```

Comment:
And here I would render it only if I have an alertText

```suggestion
      {alertText && <TemporaryAlert severity={AlertSeverity.Error} text={alertText} />}
```